### PR TITLE
fix(agw): cpp migration of SPGW files

### PR DESCRIPTION
**Pull request by pruthvihebbani**
fix(agw): cpp migration of SPGW files

## Summary

Migrated all the files under spgw task to cpp/hpp and fixed all the errors and warnings.
This PR is same as https://github.com/magma/magma/pull/13213 except that the files were renamed using mv cmd in PR #13213 . This was erasing the file commit history. In this PR git mv command is used so that the file history is retained. Review comments on PR #13213 are all incorporated in this PR as well.

## Test Plan
- Verified s1ap tester sanity
- Verified unit tests using test_oai

### CODE REVIEW

**Review by LKreutzer — APPROVED** [review]
Bazel lgtm

File: lte/gateway/c/core/oai/tasks/sgw/pgw_pco.cpp [hunk 1]
```diff
@@ -26,15 +26,21 @@
...
+#endif
+#include "lte/gateway/c/core/common/common_defs.h"
 #include "lte/gateway/c/core/oai/common/rfc_1332.h"
 #include "lte/gateway/c/core/oai/common/rfc_1877.h"
 #include "lte/gateway/c/core/oai/include/pgw_config.h"
 #include "lte/gateway/c/core/oai/include/spgw_config.h"
 #include "lte/gateway/c/core/oai/lib/3gpp/3gpp_24.008.h"
 #include "lte/gateway/c/core/oai/lib/bstr/bstrlib.h"
-#include "lte/gateway/c/core/oai/tasks/sgw/pgw_pco.h"
+#include "lte/gateway/c/core/oai/tasks/sgw/pgw_pco.hpp"
```
Comment (ajahl) [inline lte/gateway/c/core/oai/tasks/sgw/pgw_pco.cpp]: Can you move the include of the own header to the top [(Order of Includes)](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)?

File: lte/gateway/c/core/oai/tasks/sgw/pgw_procedures.cpp [hunk 2]
```diff
@@ -24,9 +24,15 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-#include "lte/gateway/c/core/common/dynamic_memory_check.h"
 #include "lte/gateway/c/core/oai/include/sgw_context_manager.h"
-#include "lte/gateway/c/core/oai/tasks/sgw/pgw_procedures.h"
+#include "lte/gateway/c/core/oai/tasks/sgw/pgw_procedures.hpp"
```
Comment (ajahl) [inline lte/gateway/c/core/oai/tasks/sgw/pgw_procedures.cpp]: The same as above [(Order of Includes)](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)

File: lte/gateway/c/core/oai/tasks/sgw/sgw_paging.cpp [hunk 3]
```diff
@@ -22,14 +22,20 @@
...
+#endif
+#include "lte/gateway/c/core/oai/include/s11_messages_types.h"
 #include "lte/gateway/c/core/oai/lib/mobility_client/MobilityClientAPI.hpp"
-#include "lte/gateway/c/core/oai/tasks/sgw/sgw_defs.h"
-#include "lte/gateway/c/core/oai/tasks/sgw/sgw_paging.h"
 #include "lte/gateway/c/core/oai/lib/itti/intertask_interface_types.h"
 #include "lte/gateway/c/core/oai/lib/itti/itti_types.h"
-#include "lte/gateway/c/core/oai/include/s11_messages_types.h"
+#include "lte/gateway/c/core/oai/tasks/sgw/sgw_defs.hpp"
+#include "lte/gateway/c/core/oai/tasks/sgw/sgw_paging.hpp"
```
Comment (ajahl) [inline lte/gateway/c/core/oai/tasks/sgw/sgw_paging.cpp]: The same as above [(Order of Includes)](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)

File: lte/gateway/c/core/oai/test/spgw_task/test_pgw_pco.cpp [hunk 4]
```diff
@@ -18,14 +18,14 @@
 #include <cstdint>
 #include <string>
 
+#include "lte/gateway/c/core/oai/tasks/sgw/pgw_pco.hpp"
```
Comment (ajahl) [inline lte/gateway/c/core/oai/test/spgw_task/test_pgw_pco.cpp]: Can you  and place it after the `extern "C"` block [(Order of Includes)](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)?

File: lte/gateway/c/core/oai/tasks/sgw/s11_causes.cpp [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *-------------------------------------------------------------------------------
+ * For more information about the OpenAirInterface (OAI) Software Alliance:
+ *      contact@openairinterface.org
+ */
+
+/*! \file s11_causes.cpp
```
Comment (ajahl) [inline lte/gateway/c/core/oai/tasks/sgw/s11_causes.cpp:18]: Can you use the git mv command for this file?

File: lte/gateway/c/core/oai/tasks/sgw/s11_causes.cpp [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+  \email: lionel.gauthier@eurecom.fr
+*/
+#define SGW
+#define S11_CAUSES_C
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+
+#include "lte/gateway/c/core/oai/tasks/sgw/s11_causes.hpp"
```
Comment (ajahl) [inline lte/gateway/c/core/oai/tasks/sgw/s11_causes.cpp]: The same as above [Order of Includes](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)

**Review by ajahl — CHANGES_REQUESTED** [review]

File: lte/gateway/c/core/oai/tasks/sgw/s11_causes.cpp (hunk 5 above)
Comment (pruthvihebbani) [reply lte/gateway/c/core/oai/tasks/sgw/s11_causes.cpp:18]: git mv cmd is only used but more than 90% lines in this file are modified. That`s the reason I guess git is not able to identify this file as renamed like other files in this PR

Comment (ajahl) [reply lte/gateway/c/core/oai/tasks/sgw/s11_causes.cpp:18]: if I use the mv command git detects it. [Draft Pull Request](https://github.com/ajahl/magma/pull/5/files)

Screenshot 2022-07-15 at 12 18 42

Comment (pruthvihebbani) [reply lte/gateway/c/core/oai/tasks/sgw/s11_causes.cpp:18]: On the console even for me it shows as renamed with git mv. But once I modify the contents it shows as deleted and added. I did this exercise separately only for this file by renaming the file(using git mv) and then modifying the contents in multiple commits here : https://github.com/magma/magma/compare/master...pruthvihebbani:magma:test_sgw_cpp_migration_s11_causes_file. But once the file modification reaches ~90% git shows it as deleted and added.

Comment (ajahl) [reply lte/gateway/c/core/oai/tasks/sgw/s11_causes.cpp:18]: Ah, I understand. Is the result the same when you rename the file in the last step (commit)?

Comment (pruthvihebbani) [reply lte/gateway/c/core/oai/tasks/sgw/s11_causes.cpp:18]: Yes the behavior is same. https://github.com/magma/magma/compare/master...pruthvihebbani:magma:test_sgw_cpp_migration_s11_causes_file_rename_last. Even though it is shown as deleted/added. The commit history is retained. So I thought it should be fine

Comment (ajahl) [reply lte/gateway/c/core/oai/tasks/sgw/s11_causes.cpp:18]: ok

**Review by ajahl — APPROVED** [review]

**Pull request merged**